packages/flutter_tools/lib/src/commands/ios.dart

```diff
@@ -48,27 +51,23 @@ class IOSCommand extends FlutterCommand {
   Future<bool> _inflateXcodeArchive(String directory, List<int> archiveBytes) async {
     print("Unzipping Xcode project to local directory...");
 
-    if (archiveBytes.isEmpty)
-      return false;
-
-    // We cannot use ArchiveFile because this archive contains file that are exectuable
+    // We cannot use ArchiveFile because this archive contains files that are exectuable
     // and there is currently no provision to modify file permissions during
-    // or after creation. See https://github.com/dart-lang/sdk/issues/15078
+    // or after creation. See https://github.com/dart-lang/sdk/issues/15078.
     // So we depend on the platform to unzip the archive for us.
 
     Directory tempDir = await Directory.systemTemp.create();
-    File tempFile = await new File(path.join(tempDir.path, "FlutterXcode.zip")).create();
-    IOSink writeSink = tempFile.openWrite();
-    writeSink.add(archiveBytes);
-    await writeSink.close();
+    File tempFile = new File(path.join(tempDir.path, "FlutterXcode.zip"))..createSync();
+    tempFile.writeAsBytesSync(archiveBytes);
 
-    ProcessResult result = await Process.run('/usr/bin/unzip', [tempFile.path, '-d', directory]);
+    try {
+      runCheckedSync(['/usr/bin/unzip', tempFile.path, '-d', directory]);
+    } catch (error) {
+      return false;
+    }
 
     // Cleanup the temp directory after unzipping
-    await Process.run('/bin/rm', ['rf', tempDir.path]);
-
-    if (result.exitCode != 0)
-      return false;
+    runSync(['/bin/rm', 'rf', tempDir.path]);
```

Comment:
Hmm, looks like we were trying to delete two files here, one called `rf`. I'll add the dash.